contracts/GenArt721CoreV3_Engine_Flex.sol

```diff
@@ -0,0 +1,2278 @@
+            .dependencyType = _dependencyType;
+        // if the incoming dependency type is onchain, we need to write the data to bytecode
+        if (_dependencyType == ExternalAssetDependencyType.ONCHAIN) {
+            if (oldDependencyType != ExternalAssetDependencyType.ONCHAIN) {
+                // we only need to set the cid to an empty string if we are replacing an offchain asset
+                // an onchain asset will already have an empty cid
+                projects[_projectId].externalAssetDependencies[_index].cid = "";
+            }
+
+            projects[_projectId]
+                .externalAssetDependencies[_index]
+                .bytecodeAddress = _cidOrData.writeToBytecode();
+            // we don't want to emit data, so we emit the cid as an empty string
+            _cidOrData = "";
+        } else {
+            projects[_projectId]
+                .externalAssetDependencies[_index]
+                .cid = _cidOrData;
+        }
+        emit ExternalAssetDependencyUpdated(
+            _projectId,
+            _index,
+            _cidOrData,
+            _dependencyType,
+            assetCount
+        );
+    }
+
+    /**
+     * @notice Removes external asset dependency for project `_projectId` at index `_index`.
+     * Removal is done by swapping the element to be removed with the last element in the array, then deleting this last element.
+     * Assets with indices higher than `_index` can have their indices adjusted as a result of this operation.
+     * @param _projectId Project to be updated.
+     * @param _index Asset index
+     */
+    function removeProjectExternalAssetDependency(
+        uint256 _projectId,
+        uint256 _index
+    ) external {
+        _onlyUnlockedProjectExternalAssetDependencies(_projectId);
+        _onlyArtistOrAdminACL(
+            _projectId,
+            this.updateProjectExternalAssetDependency.selector
+        );
+        uint24 assetCount = projects[_projectId].externalAssetDependencyCount;
+        require(_index < assetCount, "Asset index out of range");
+
+        uint24 lastElementIndex = assetCount - 1;
+
+        // copy last element to index of element to be removed
+        projects[_projectId].externalAssetDependencies[_index] = projects[
+            _projectId
+        ].externalAssetDependencies[lastElementIndex];
+
+        if (
+            projects[_projectId]
+                .externalAssetDependencies[lastElementIndex]
+                .dependencyType == ExternalAssetDependencyType.ONCHAIN
+        ) {
+            // copy last element bytecode address to index of element to be removed
+            // if we are dealing with an onchain external asset dependency
+            projects[_projectId]
+                .externalAssetDependencies[_index]
+                .bytecodeAddress = projects[_projectId]
+                .externalAssetDependencies[lastElementIndex]
+                .bytecodeAddress;
+        }
```

Comment:
I believe this operation should already should have been done by the previous statement?